assignment-2.md

```diff
@@ -0,0 +1,62 @@
+# Assignment 2: Render Dynamic Data
+
+Take a look at the existing `Climate` component. It receives a `Sensor` as a
+prop. This sensor is an `EventEmitter` that emits `temperature` and `humidity`
+events. The component is registering event handlers to those, and updates its
+state accordingly. This leads to React rendering the component each time a new
+value is received.
+
+## 2.1 Extract Child Component
+
+The existing `Climate` component has some duplication in it: event handling and
+rendering of the emitted values is the same for *temperature* and *humidity*.
+Being good developers, we are aware of the
+[*single responsibility principle*](https://en.wikipedia.org/wiki/Single-responsibility_principle)
+and [*DRY*](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself), so let's
+remove that. We will see later that this makes the addition of new features much
+easier!
+
+Introduce a new component that can be used twice by the `Climate` component, to
+deal with either *temperature* or *humidity*.
+
+See what that component needs from its parent, and pass those things via props.
+The `render` method of `Climate` will use the new component twice: once for the
+temperature, and once for the humidity.
+
+### Hints
+
+This is a refactoring task, so the functionality should not change! You can even
+use the existing test (run with `npm test`) to be sure.
+
+## 2.2 Add Minimum and Maximum
+
+In addition to current temperature and humidity, the minimum and maximum of
+those values over time shall be shown.
+
+Adapt your new component so that it keeps track of the minimum and maximum in
```

Comment:
The assignment "drives" people to keep track of the state inside the child components, I don't think we should push them in that direction that explicitly. 
I think it's better to let them come up with their own solution. Maybe we have more than one approach, which we can talk about later one in the assignment wrap-up.

(also I think keeping the state in the parent is a better solution)

Reply:
You are right. I've shortened it to just the task to be done, without given any impl. details:
https://github.com/methodpark/react-workshop/compare/5c9e58f347d833dcd2c74f479fab3dd59f45e9a4..d7ae133c42e27af4319fbad4ccd1ffde3a3cfb74